src/Documents/RecordsDocument.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Terraformers\OpenArchive\Documents;
+
+use Exception;
+use SilverStripe\ORM\PaginatedList;
+use Terraformers\OpenArchive\Formatters\OaiRecordFormatter;
+use Terraformers\OpenArchive\Helpers\ResumptionTokenHelper;
+use Terraformers\OpenArchive\Models\OaiRecord;
+
+abstract class RecordsDocument extends OaiDocument
+{
+
+    protected ?OaiRecordFormatter $formatter = null;
+
+    abstract public function shouldIncludeMetadata(): bool;
+
+    abstract public function getRecordsHolderName(): string;
+
+    public function __construct(?OaiRecordFormatter $formatter = null)
+    {
+        parent::__construct();
+
+        if (!$formatter) {
+            return;
+        }
+
+        $this->setMetadataPrefix($this->formatter->getMetadataPrefix());
+        $this->formatter = $formatter;
```

Comment:
```suggestion
        $this->formatter = $formatter;
        $this->setMetadataPrefix($this->formatter->getMetadataPrefix());
```
Can we please switch the order of these two lines of code? The value `$this->formatter` is `null` before we assign it a formatter.